### PR TITLE
Update kustomize files for new Dataplane CRD design

### PR DESCRIPTION
**Pull request by bshephar**
Testing the CRD Proposal PR.
https://github.com/openstack-k8s-operators/dataplane-operator/pull/303

Need to baseline this against a CI environment with both Baremetal and Preprovisioned nodes.

Related to: https://github.com/openstack-k8s-operators/ci-framework/pull/479

### CODE REVIEW

**Comment by bshephar** [comment]
recheck

**Comment by bshephar** [comment]
recheck

**Comment by bshephar** [comment]
Hmm, seems to be using the default repo still:
```
2023-08-24 06:15:35.325892 | controller |         "DATAPLANE_REPO": "/home/zuul/src/github.com/openstack-k8s-operators/dataplane-operator",
```

**Comment by bshephar** [comment]
recheck

File: Makefile [hunk 1]
```diff
@@ -281,11 +281,11 @@ BAREMETAL_BRANCH    ?= main
...
-DATAPLANE_IMG                                    ?= quay.io/openstack-k8s-operators/dataplane-operator-index:latest
-DATAPLANE_REPO                                   ?= https://github.com/openstack-k8s-operators/dataplane-operator.git
-DATAPLANE_BRANCH                                 ?= main
-OPENSTACK_DATAPLANE                              ?= config/samples/dataplane_v1beta1_openstackdataplane_with_ipam.yaml
-OPENSTACK_DATAPLANE_BAREMETAL                    ?= config/samples/dataplane_v1beta1_openstackdataplane_baremetal_with_ipam.yaml
+DATAPLANE_IMG                                    ?= quay.io/bshephar/dataplane-operator-index:v0.0.1
+DATAPLANE_REPO                                   ?= https://github.com/bshephar/dataplane-operator.git
+DATAPLANE_BRANCH                                 ?= crd-proposal
+OPENSTACK_DATAPLANE                              ?= config/samples/dataplane_v1beta1_openstackdataplanenodeset.yaml
+OPENSTACK_DATAPLANE_BAREMETAL                    ?= config/samples/dataplane_v1beta1_openstackdataplanenodeset_baremetal_with_ipam.yaml
```
[the remark this answers is not in the document]
Comment (bshephar) [reply Makefile]: Can do. Changed the others as well.

**Comment by slagle** [comment]
recheck

**Comment by bshephar** [comment]
Hmm still trying to checkout the OpenStack-k8s-operators version of dataplane-operator which doesn’t have the crd-proposal branch. Weird that the branch part works but the DATAPLANE_REPO env var doesn’t. Must be missing something somewhere here. Possibly in ci-framework rather than install_yamls though.

**Comment by fao89** [comment]
needs https://github.com/openstack-k8s-operators/openstack-operator/pull/453 to catch up

**Review by fao89 — APPROVED** [review]

**Pull request merged**